### PR TITLE
Support FES same day indicator for form templates

### PR DESCRIPTION
**Pull request by hepsimo**
- update API spec; add form template property isSameDayService
- update API spec; change /form-template path param to query param
- use FormTemplateApi property isSameDay()
- apply builder pattern to FormModel
- apply builder pattern to FormTemplate
- add SH19_SAMEDAY to form_templates.json
- include unit test coverage

Resolves:
BI-7848

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/efs/api/events/service/model/FesLoaderModel.java [hunk 1]
```diff
@@ -12,23 +14,29 @@
     private String companyName;
     private String companyNumber;
     private String formType;
+    private boolean sameDay;
 
     public FesLoaderModel(String barcode, String companyName, String companyNumber, String formType,
```
Comment (PhilipGlover) [inline src/main/java/uk/gov/companieshouse/efs/api/events/service/model/FesLoaderModel.java:19]: the PR description says that a builder has been added to this class.  Am I looking in the right place?

**Review by PhilipGlover — COMMENTED** [review]
I've run the code and tested by changing the category of the `SH19_SAMEDAY` formTemplate, so that it does list on screen.  I then selected and attempted to submit to FES.  That doesn't work locally cos we haven't got EFS doc processor running locally to convert the files to tif.  But looking at debug all is as expected.
I'm ready to approve once:
- sdk version updated (I've approved that PR)
- we then hopefully see that the analyse-PR job pass in concourse
- minor discrepancy re Builder pattern between PR description and code resolved

File: src/main/java/uk/gov/companieshouse/efs/api/events/service/model/FesLoaderModel.java (hunk 1 above)
Comment (hepsimo) [reply src/main/java/uk/gov/companieshouse/efs/api/events/service/model/FesLoaderModel.java:19]: Good spot, it should have said FormTemplate. I'll fix the description

**Review by PhilipGlover — APPROVED** [review]
as per my earlier comment - happy to approve now

**Pull request merged**